### PR TITLE
occ commands to manage logging

### PR DESCRIPTION
**Pull request by RobinMcCorkell**
In similar vein to the maintenance:\* commands and others, this PR introduces two new occ commands, log:manage and log:owncloud. It takes away the requirement to look at the documentation to change logging settings.

log:manage allows the following information to be set/displayed: logging backend (owncloud, errorlog, syslog), log level and log timezone.

log:owncloud manages the owncloud logging backend, and can be used to set/display: log filename and log rotation size. These are parameters specific to the owncloud backend, and so get their own command

@PVince81 we discussed this briefly

cc @DeepDiver1975 @MorrisJobke @LukasReschke

### CODE REVIEW

**Comment by karlitschek** [comment]
Very cool and useful!! :+1:

**Comment by Xenopathic** [comment]
It'd be nice if the CI build slave didn't shutdown while testing this... I'll try again in the morning.

**Comment by MorrisJobke** [comment]
@owncloud-bot retest this please

**Comment by DeepDiver1975** [comment]
````
02:33:57 + phpenv local 5.3.28
02:33:57 /tmp/hudson6519004932099214154.sh: 3: /tmp/hudson6519004932099214154.sh: phpenv: not found

**Comment by DeepDiver1975** [comment]
@DeepDiver1975 STOP PLAYING AROUND WITH THE SI SYSTEM!

**Comment by DeepDiver1975** [comment]
@owncloud-bot retest this please

**Comment by DeepDiver1975** [comment]
@owncloud-bot retest this please

**Comment by DeepDiver1975** [comment]
@owncloud-bot restest this please

File: core/command/log/manage.php [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+				throw new \InvalidArgumentException('Invalid backend');
+			}
+			$toBeSet['logtype'] = $backend;
+		}
+
+		if ($level = $input->getOption('level')) {
+			if (is_numeric($level)) {
+				$levelNum = $level;
+				// sanity check
+				$level = $this->convertLevelNumber($levelNum);
```
Comment (DeepDiver1975) [inline core/command/log/manage.php]: $level is unused

File: core/command/log/manage.php [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+	/**
+	 * @param string $level
+	 * @return int
+	 */
+	protected function convertLevelString($level) {
+		$level = strtolower($level);
+		switch ($level) {
+		case 'debug':
+			return 0;
+			break;
```
Comment (DeepDiver1975) [inline core/command/log/manage.php]: no break after return - unreachable code

File: core/command/log/owncloud.php [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+
+	public function __construct(IConfig $config) {
+		$this->config = $config;
+		parent::__construct();
+	}
+
+	protected function configure() {
+		$this
+			->setName('log:owncloud')
+			->setDescription('manipulate owncloud logging backend')
```
Comment (DeepDiver1975) [inline core/command/log/owncloud.php]: ownCloud

File: core/command/log/owncloud.php [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+				'file',
+				null,
+				InputOption::VALUE_REQUIRED,
+				'set the log file, relative to data'
+			)
+			->addOption(
+				'rotate-size',
+				null,
+				InputOption::VALUE_REQUIRED,
+				'set the filesize for log rotation, 0 = disabled'
```
Comment (DeepDiver1975) [inline core/command/log/owncloud.php]: file size

File: core/command/log/owncloud.php [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+			$this->config->setSystemValue($option, $value);
+		}
+
+		// display config
+		if ($this->config->getSystemValue('logtype', 'owncloud') === 'owncloud') {
+			$enabledText = 'enabled';
+		} else {
+			$enabledText = 'disabled';
+		}
+		$output->writeln('Log backend owncloud: '.$enabledText);
```
Comment (DeepDiver1975) [inline core/command/log/owncloud.php]: ownCloud

**Comment by DeepDiver1975** [comment]
I'd like to see unit test on these new commands - THX

**Comment by Xenopathic** [comment]
@owncloud-bot retest this please

File: config/config.sample.php [hunk 6]
```diff
@@ -493,9 +493,10 @@
 'log_type' => 'owncloud',
 
 /**
- * Change the ownCloud logfile name from ``owncloud.log`` to something else.
```
Comment (DeepDiver1975) [inline config/config.sample.php]: @carlaschroder please review these changes since the will be re-used in the documentation

**Comment by PVince81** [comment]
:+1:

Now all I'm missing is another command "./occ log:human" that outputs the log in a properly formatted way, with nicely presented stack traces (I'm tired of manually reformatting stack traces in bug reports)

**Comment by Xenopathic** [comment]
@PVince81 Getting there :smile:

**Comment by MorrisJobke** [comment]
Tested and works :+1:

**Pull request merged**